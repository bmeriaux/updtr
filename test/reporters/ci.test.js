import EventEmitter from "events";
import { WritableStreamBuffer } from "stream-buffers";
import unicons from "unicons";
import ci from "../../src/reporters/ci";
import events from "../fixtures/events";

function setup(reporterConfig = {}) {
    const updtr = new EventEmitter();
    const stdout = new WritableStreamBuffer();

    stdout.isTTY = true;
    stdout.columns = 80;

    reporterConfig.stream = stdout;
    ci(updtr, reporterConfig);

    return {
        updtr,
        stdout,
    };
}

beforeAll(() => {
    // We need to replace platform-dependent characters with static counter parts to make snapshot testing
    // consistent across platforms.
    unicons.cli = sign => {
        switch (sign) {
            case "circle":
                return "-";
            default:
                return "";
        }
    };
});

describe("ci()", () => {
    Object.keys(events).forEach(caseName => {
        describe(caseName, () => {
            it("should print the expected lines", async () => {
                const testCase = events[caseName];
                const { updtr, stdout } = setup(testCase.reporterConfig);

                await testCase.events.reduce(async (previous, [
                    eventName,
                    event,
                ]) => {
                    await previous;
                    updtr.emit(eventName, event);

                    // Faking async events
                    return Promise.resolve();
                }, Promise.resolve());

                const output = stdout.getContentsAsString("utf8");

                expect(
                    output.replace(
                        // We need to replace the timing because that is non-deterministic
                        /Finished after \d+\.\ds/,
                        "Finished after 1.0s"
                    )
                ).toMatchSnapshot(caseName);
            });
        });
    });
});
