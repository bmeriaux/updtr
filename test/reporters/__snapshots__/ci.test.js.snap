// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom config and only excluded modules 1`] = `
"[?25l[JRunning updtr with custom configuration:

- exclude: b, c

[JInstalling missing dependencies[90m...[39m
[90m> npm install [39m
[JLooking for outdated modules[90m...[39m
[90m> npm outdated [39m
[J[1mNo updates available[22m for the given modules and version range
[J
[J[1m3 skipped[22m modules:

[90m-[39m [1ma[22m [90mgit[39m
[90m-[39m [1mb[22m [90mexcluded[39m
[90m-[39m [1mc[22m [90mexcluded[39m
[J
Finished after 1.0s.
"
`;

exports[`no outdated modules 1`] = `
"[?25l[JInstalling missing dependencies[90m...[39m
[90m> npm install [39m
[JLooking for outdated modules[90m...[39m
[90m> npm outdated [39m
[JEverything [1mup-to-date[22m
[J
[J
Finished after 1.0s.
"
`;
